pkg/platform/provider/baremetal/phases/kubeadm/kubeadm.go

```diff
@@ -360,6 +383,25 @@ func UpgradeNode(s ssh.Interface, client kubernetes.Interface, platformClient pl
 	return true, nil
 }
 
+func checkKubeletVersion(client kubernetes.Interface, nodeName, version string, ignorePatchVersion bool) (same bool, err error) {
+	node, err := client.CoreV1().Nodes().Get(context.TODO(), nodeName, metav1.GetOptions{})
+	if err != nil {
+		return false, err
+	}
+	if ignorePatchVersion {
+		same, err = sameMinorVersion(node.Status.NodeInfo.KubeletVersion, version)
```

Comment:
please merge sameMinorVersion your logic to sameVersion so that keep one function to caller.